step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/aggregated/ReportNodeTimeSeries.java

```diff
@@ -0,0 +1,64 @@
+package step.core.artefacts.reports.aggregated;
+
+import step.core.artefacts.reports.ReportNode;
+import step.core.artefacts.reports.ReportNodeStatus;
+import step.core.collections.CollectionFactory;
+import step.core.collections.Filters;
+import step.core.collections.IndexField;
+import step.core.collections.Order;
+import step.core.collections.filters.And;
+import step.core.timeseries.TimeSeries;
+import step.core.timeseries.TimeSeriesIngestionPipeline;
+import step.core.timeseries.aggregation.TimeSeriesAggregationPipeline;
+import step.core.timeseries.aggregation.TimeSeriesAggregationQueryBuilder;
+import step.core.timeseries.bucket.BucketAttributes;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ReportNodeTimeSeries implements Closeable {
```

Comment:
Could/should this even be AutoCloseable? (sincere question, I don't really know if that has drawbacks/benefits).

Reply:
It is autocloesable (Closeable extends AutoCloseable) and can be used in a try with resource construct. However the time  serie is created and added to the Step controller context and available during the lifetime of  the controller. The Step context when closed will automatically close all stored object which are instance of and only of Closeable